_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,165 @@
+
+We have the source code, so we can describe what to do with this code. As I mentioned before we will **test** our application and generate an **artifact** with executable content.   
+
+**TeamCity** has predefined runners for different build tools. In this case we are going to use the **Gradle** runner to **test** and **build**. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_step.png" alt="Gradle Step" title="Gradle Step" class="img img-responsive style-screengrab">
+
+Now we can run our build. As result of we should see our **artifacts** in *"Artifact"* tab.
+ 
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_result.png" alt="Build Result" title="Build Result" class="img img-responsive style-screengrab">
+
+## Release
+
+Our code is no longer needed. We have all we need to build the docker image. Now we have to create our image and **release** with the right version. To simplify our example we are going to use the current build number to define an image version.
+Next we will generate a file with this version. This file give us the possibility to pass information about the version to the next steps.  
+
+First, let's take a closer look at the `Dockerfile`. We copy all libraries and scripts from `simple_application.tar` to the image. This ***tar*** file already contains all required content, so all we have to do is to use `ADD`. It will automatically *untar* all files in the image. Next we expose the port of our HTTP API and we define how we launch our application by adding an `ENTRYPOINT` command.  
+
+For example our `Dockerfile` can look like this: 
+
+```
+FROM java:8
+
+ADD simple_application.tar .
+
+EXPOSE 4567
+
+ENTRYPOINT ["/simple_application/bin/simple_application"]
+```
+
+Defining the build configuration is very simple. In the general settings we just define a new artifact: `image.version`. We will generate this file in a build step and it will contain a docker image version.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_config.png" alt="Release Configuration" title="Release Configuration" class="img img-responsive style-screengrab">
+
+Without the **artifact** we won't be able to build any image. We have to tell our build how to find **artifacts** from the **Build** phase. We can do that by defining an **Artifact Dependency** in **TeamCity**. We just have to choose a build configuration and define files which we want to use in this build.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_artifacts.png" alt="Release Artifact Dependency" title="Release Artifact Dependency" class="img img-responsive style-screengrab">
+
+And finally we have to trigger this build automatically when we pass all **tests** in the previous step. By introducing **Finish Build Trigger** we can start this build just after **TeamCity** successfully finishes building the application.  
```

Comment:
- And finally we have to trigger this build after all tests triggered by the previous step pass.